### PR TITLE
SidebarNav

### PR DESCRIPTION
**Pull request by jivey**
https://openstax.atlassian.net/browse/DISCO-29

### CODE REVIEW

File: src/components/SidebarNav.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  @keyframes expandSidebarNav {
+    from {
+      width: var(--collapsed-width);
+    }
+    to {
+      width: var(--expanded-width);
+    }
+  }
+
+  @keyframes collapseSidebarNav {
```
Comment (bethshook) [inline src/components/SidebarNav.tsx:66]: why was animation necessary vs a transition?

**Comment by bethshook** [comment]
looks good in ladle, just left a question

**Review by bethshook — APPROVED** [review]

File: src/components/SidebarNav.tsx (hunk 1 above)
Comment (jivey) [reply src/components/SidebarNav.tsx:66]: For mobile, I needed to remount it under a Modal, so the transition doesn't get a chance to play. This fakes the transition, running the expand on mount inside the modal and the collapse on normal mount. Thankfully it still worked too for the desktop case, though I'm realizing now I probably could have kept the transition for that case and kept the keyframe stuff only for mobile...

Comment (bethshook) [reply src/components/SidebarNav.tsx:66]: ah of course, that makes sense

**Pull request merged**